azkaban-web-server/src/main/java/azkaban/imagemgmt/services/ImageVersionServiceImpl.java

```diff
@@ -72,6 +77,32 @@ public int createImageVersion(ImageMetadataRequest imageMetadataRequest)
   @Override
   public List<ImageVersion> findImageVersions(ImageMetadataRequest imageMetadataRequest)
       throws ImageMgmtException {
-    return imageVersionsDao.findImageVersions(imageMetadataRequest);
+    Set<String> imageTypes = new LinkedHashSet<>();
+    imageTypes.add("azkaban_core");
+    imageTypes.add("azkaban_exec");
+    imageTypes.add("hadoop_java_job");
+    imageTypes.add("kabootar_job");
+    imageTypes.add("wormhole_job");
+    return imageVersionsDao.getActiveVersionByImageTypes(imageTypes);
+    //return imageVersionsDao.findImageVersions(imageMetadataRequest);
+  }
+
+  @Override
+  public void updateImageVersion(ImageMetadataRequest imageMetadataRequest) throws IOException,
+      ImageMgmtException {
+    // Convert input json payload to image version update request object
+    ImageVersionRequest imageVersionRequest =
+        converterUtils.convertToModel(imageMetadataRequest.getJsonPayload(),
+            ImageVersionRequest.class);
+    // Set the user who invoked the update API
+    imageVersionRequest.setModifiedBy(imageMetadataRequest.getUser());
+    imageVersionRequest.setId((Integer) imageMetadataRequest.getParams().get(ID_KEY));
+    log.info("imageVersionUpdateRequest: " + imageVersionRequest);
+    // Input validation for image version create request
+    if (!ValidatorUtils.validateObject(imageVersionRequest)) {
+      throw new ImageMgmtValidationException("Provide valid input for updating image version "
+          + "metadata");
+    }
```

Comment:
We should provide concrete error messages, be specific about the parameters that are invalid.

Reply:
Done.